WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift

```diff
@@ -0,0 +1,273 @@
+                         .paragraphStyle: paragraph]
+        )
+
+        mutableAttributedText.setAsLink(textToFind: Localization.termsOfService,
+                                        linkURL: Constants.termsLink)
+        mutableAttributedText.setAsLink(textToFind: Localization.adPolicy,
+                                        linkURL: Constants.adPolicyLink)
+        mutableAttributedText.setAsLink(textToFind: Localization.learnMore,
+                                        linkURL: Constants.learnMoreLink)
+        return mutableAttributedText
+    }()
+
+    init(viewModel: BlazeConfirmPaymentViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        ScrollView {
+            VStack(spacing: Layout.contentPadding) {
+
+                totalAmountView
+                    .padding(.horizontal, Layout.contentPadding)
+
+                Divider()
+
+                if !viewModel.isFetchingPaymentInfo {
+                    if viewModel.selectedPaymentMethod == nil {
+                        addPaymentMethodButton
+                            .padding(.horizontal, Layout.contentPadding)
+                    } else {
+                        cardDetailView
+                            .padding(.horizontal, Layout.contentPadding)
+                    }
+
+                } else {
+                    loadingView
+                        .padding(.horizontal, Layout.contentPadding)
+                }
+
+                Divider()
+            }
+            .padding(.vertical, Layout.contentPadding)
+        }
+        .navigationTitle(Localization.title)
+        .navigationBarTitleDisplayMode(.inline)
+        .safeAreaInset(edge: .bottom) {
+            footerView
+        }
+        .task {
+            await viewModel.updatePaymentInfo()
+        }
+    }
+}
+
+private extension BlazeConfirmPaymentView {
+    var totalAmountView: some View {
+        VStack(alignment: .leading, spacing: Layout.contentPadding) {
+            Text(Localization.paymentTotals)
+                .bold()
+                .bodyStyle()
+
+            HStack {
+                Text(Localization.blazeCampaign)
+                    .bodyStyle()
+
+                Spacer()
+
+                Text(viewModel.totalAmount)
+            }
+            .frame(maxWidth: .infinity)
+
+            HStack {
+                Text(Localization.total)
+                    .bold()
+
+                Spacer()
+
+                Text(String.localizedStringWithFormat(Localization.totalAmount, viewModel.totalAmount))
+                    .bold()
+            }
+            .bodyStyle()
+        }
+    }
+
+    var cardDetailView: some View {
+        Button {
+            // TODO: show payment method list
+        } label: {
+            if let icon = viewModel.cardIcon {
+                Image(uiImage: icon)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 35 * scale)
```

Comment:
nit: Should we move the width value to Layout enum?

Reply:
Done in efcd530a05064b1c588d4c7d5192e6de773c66bf.